src/Bridges/Constraint/geomean_to_relentr.jl

```diff
@@ -0,0 +1,92 @@
+    return GeoMeantoRelEntrBridge{T, F, G, H}
+end
+
+# Attributes, Bridge acting as a model
+MOI.get(bridge::GeoMeantoRelEntrBridge, ::MOI.NumberOfVariables) = 1
+MOI.get(bridge::GeoMeantoRelEntrBridge, ::MOI.ListOfVariableIndices) = [bridge.y]
+MOI.get(bridge::GeoMeantoRelEntrBridge{T, F}, ::MOI.NumberOfConstraints{F, MOI.GreaterThan{T}}) where {T, F} = 1
+MOI.get(bridge::GeoMeantoRelEntrBridge{T, F, G}, ::MOI.NumberOfConstraints{G, MOI.RelativeEntropyCone}) where {T, F, G} = 1
+MOI.get(bridge::GeoMeantoRelEntrBridge{T, F}, ::MOI.ListOfConstraintIndices{F, MOI.GreaterThan{T}}) where {T, F} = [bridge.ge_index]
+MOI.get(bridge::GeoMeantoRelEntrBridge{T, F, G}, ::MOI.ListOfConstraintIndices{G, MOI.RelativeEntropyCone}) where {T, F, G} = [bridge.relentr_index]
+
+# References
+function MOI.delete(model::MOI.ModelLike, bridge::GeoMeantoRelEntrBridge)
+    MOI.delete(model, bridge.relentr_index)
+    MOI.delete(model, bridge.ge_index)
+    MOI.delete(model, bridge.y)
+end
+
+# Attributes, Bridge acting as a constraint
+function MOI.get(model::MOI.ModelLike, ::MOI.ConstraintFunction, bridge::GeoMeantoRelEntrBridge{T, F, G, H}) where {T, F, G, H}
+    relentr_func = MOIU.eachscalar(MOI.get(model, MOI.ConstraintFunction(), bridge.relentr_index))
+    d = div(length(relentr_func) - 1, 2)
+    u_func = MOIU.remove_variable(MOIU.operate(-, T, relentr_func[end], MOI.SingleVariable(bridge.y)), bridge.y)
+    w_func = relentr_func[2:(1 + d)]
+    return MOIU.convert_approx(H, MOIU.operate(vcat, T, u_func, w_func))
+end
+MOI.get(model::MOI.ModelLike, ::MOI.ConstraintSet, bridge::GeoMeantoRelEntrBridge) = MOI.GeometricMeanCone(1 + div(MOI.dimension(MOI.get(model, MOI.ConstraintSet(), bridge.relentr_index)) - 1, 2))
+MOI.supports(::MOI.ModelLike, ::Union{MOI.ConstraintPrimalStart, MOI.ConstraintDualStart}, ::Type{<:GeoMeantoRelEntrBridge}) = true
+function MOI.get(model::MOI.ModelLike, attr::Union{MOI.ConstraintPrimal, MOI.ConstraintPrimalStart}, bridge::GeoMeantoRelEntrBridge)
+    relentr_primal = MOI.get(model, attr, bridge.relentr_index)
+    d = div(length(relentr_primal) - 1, 2)
+    y_val = MOI.get(model, attr, bridge.ge_index)
+    u_val = sum(relentr_primal[(2 + d):end]) / d - y_val
+    return vcat(u_val, relentr_primal[2:(1 + d)])
+end
+function MOI.set(model::MOI.ModelLike, attr::MOI.ConstraintPrimalStart, bridge::GeoMeantoRelEntrBridge, value)
+    u_val = value[1]
+    u_pos = -min(zero(u_val), u_val)
+    d = length(value) - 1
+    MOI.set(model, MOI.VariablePrimalStart(), bridge.y, u_pos)
+    MOI.set(model, attr, bridge.ge_index, u_pos)
+    MOI.set(model, attr, bridge.relentr_index, vcat(0, value[2:end], fill(u_pos + u_val, d)))
+    return
+end
+# Given a is dual on y >= 0 and (b, c, d) is dual on RelativeEntropyCone constraint,
+# dual on (u, w) in GeometricMeanCone is (-a, c)
+function MOI.get(model::MOI.ModelLike, attr::Union{MOI.ConstraintDual, MOI.ConstraintDualStart}, bridge::GeoMeantoRelEntrBridge)
```

Comment:
Here is how I get the dual:
A bridge can always be represented as a transformation from an F-in-S constraint into creating auxiliary vars y and constraints in F1-in-S1, F2-in-S2, ...
So you have:
x in S <=> there exists y such that Ax + By in S1 x S2 x ... x Sn
Theorem:
[I; 0] * S* = [A'; B'] (S1 x S2 x ... x Sn)*
So you have
S* = A' * (S1 x S2 x ... x Sn)*
and
0 = B' * (S1 x S2 x ... x Sn)* -> This is in fact the dual constraint due to the fact that y is free
In this case, you have S1 = RelEntr and S2 = R+
A = [0 0; 0 I; 0 e; 0 0]
B = [0; 0; e; 1]
So given a dual z = [b; c; d; a], you get
A' * z = [e' * d; c]
B' * z = e' * d + a
The first equation says that what you are looking for is [e' * d; c]
The second says that e' * d = -a so the (-a, c) you are using is indeed correct

Reply:
just amending: A = [0 0; 0 I; e 0; 0 0].
and for extra details: x = (u, w).

Reply:
thanks for this very helpful explanation @blegat